### PR TITLE
Alerting: Skip query cache for alert queries

### LINKED ISSUE #34146: Send `X-Cache-Skip` header on Alert queries
**What would you like to be added**:

We should be sending a header with queries for alerting that causes the cache middleware in Grafana Enterprise to be skipped.

The header is `X-Cache-Skip`, and can have any value. As long as it's in the request, the middleware will short circuit and continue up the chain.

**Why is this needed**:

Alert queries should probably skip the data source cache in Grafana Enterprise.

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**: 

This adds the [X-Cache-Skip](https://grafana.com/docs/grafana/latest/enterprise/query-caching/#sending-a-request-without-cache) header for alert queries.

**Which issue(s) this PR fixes**:

Fixes #34146

**Special notes for your reviewer**:

Am I missing any other places where this should be set?

### CODE REVIEW

**Review by kminehart — APPROVED** [review]

**Comment by kminehart** [comment]
> Am I missing any other places where this should be set?

Would be a good question for @kylebrandt or @marefr I think.

**Comment by kminehart** [comment]
Ahh, hold up @toddtreece the header is actually

`X-Cache-Skip`

File: pkg/services/ngalert/eval/eval.go [hunk 1]
```diff
@@ -124,7 +124,8 @@ func GetExprRequest(ctx AlertExecCtx, data []models.AlertQuery, now time.Time) (
 		OrgId: ctx.OrgID,
 		Headers: map[string]string{
 			// Some data sources check this in query method as sometimes alerting needs special considerations.
-			"FromAlert": "true",
+			"FromAlert":    "true",
+			"X-Skip-Cache": "true",
```
Comment (kminehart) [inline pkg/services/ngalert/eval/eval.go:128]:
```suggestion
			"X-Cache-Skip": "true",
```

**Review by kminehart — CHANGES_REQUESTED** [review]

File: pkg/services/ngalert/eval/eval.go (hunk 1 above)
Comment (toddtreece) [reply pkg/services/ngalert/eval/eval.go]: thanks. it looks like the other one from yesterday is fine, but i need ☕ this morning. fixed with 0522aad

**Review by kminehart — APPROVED** [review]

**Review by kylebrandt — APPROVED** [review]
Thanks!

**Pull request merged**